web/src/components/FilingsCalendarTaxAccess.test.tsx

```diff
@@ -0,0 +1,338 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
+import { generateProfileData, generateUserData, randomLegalStructure } from "@/test/factories";
+import { markdownToText, randomElementFromArray } from "@/test/helpers";
+import {
+  currentUserData,
+  setupStatefulUserDataContext,
+  WithStatefulUserData,
+} from "@/test/mock/withStatefulUserData";
+import { FormationLegalType } from "@businessnjgovnavigator/shared/index";
+import { randomInt } from "@businessnjgovnavigator/shared/intHelpers";
+import { OperatingPhases } from "@businessnjgovnavigator/shared/operatingPhase";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { BusinessPersona } from "../../../shared/src/profileData";
+import { generateFormationData } from "../../test/factories";
+import { FilingsCalendarTaxAccess } from "./FilingsCalendarTaxAccess";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({
+  postTaxRegistrationOnboarding: jest.fn(),
+}));
+const mockApi = api as jest.Mocked<typeof api>;
+
+const Config = getMergedConfig();
+
+const renderFilingsCalendarTaxAccess = (initialUserData?: UserData) => {
+  render(
+    <WithStatefulUserData initialUserData={initialUserData}>
+      <FilingsCalendarTaxAccess />
+    </WithStatefulUserData>
+  );
+};
+
+describe("<FilingsCalendarTaxAccess />", () => {
+  let userData: UserData;
+  let flow: Exclude<BusinessPersona, undefined>;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    const legalStructure = randomLegalStructure(true);
+    mockApi.postTaxRegistrationOnboarding.mockImplementation(() =>
+      Promise.resolve({ ...userData, taxFilingData: { ...userData.taxFilingData, state: "SUCCESS" } })
+    );
+    userData = generateUserData({
+      profileData: generateProfileData({
+        legalStructureId: legalStructure.id,
+        operatingPhase: randomElementFromArray(
+          OperatingPhases.filter((obj) => obj.displayTaxAccessButton === true)
+        ).id,
+      }),
+      formationData: generateFormationData(
+        { completedFilingPayment: false },
+        legalStructure.id as FormationLegalType
+      ),
+    });
+    flow = userData.profileData.businessPersona ?? flow;
+    setupStatefulUserDataContext();
+  });
+
+  it("opens Gov2Go modal when on button click", () => {
+    renderFilingsCalendarTaxAccess(userData);
+    fireEvent.click(screen.getByTestId("get-tax-access"));
+    expect(screen.getByTestId("modal-content")).toBeInTheDocument();
+  });
+
+  it("Pre-populates fields with userData values", async () => {
+    renderFilingsCalendarTaxAccess({
+      ...userData,
+      profileData: { ...userData.profileData, taxId: "123456789123", businessName: "1234" },
```

Comment:
not to nitpick, but often, it can make tests more readable if the test values we use are in line with what people expect them to be. Using "1234" as a business name is a little unexpected because it's a number, not a name. It may cause a double take. Especially when used next to "real" numeric field like taxId. Using anything else that's actually a word, not a number, might increase readability